services/cargohold/src/CargoHold/Run.hs

```diff
@@ -93,6 +97,15 @@ mkApp o = Codensity $ \k ->
             :<|> hoistServerWithDomain @InternalAPI (toServantHandler e) internalSitemap
         )
         r
+        cont
+
+    lookupReqId :: Logger -> Wai.Request -> IO RequestId
+    lookupReqId l r = case lookup requestIdName $ Wai.requestHeaders r of
+      Just rid -> pure $ RequestId rid
+      Nothing -> do
+        localRid <- RequestId . cs . UUID.toText <$> UUID.nextRandom
+        Log.info l $ "request-id" .= localRid ~~ "request" .= (show r) ~~ msg (val "generated a new request id for local request")
```

Comment:
Same as brig, probably better to log individual items from the request rather than the whole request.